packages/module/src/pipelines/components/nodes/TaskNode.tsx

```diff
@@ -303,16 +304,25 @@ const TaskNodeInner: React.FC<TaskNodeInnerProps> = observer(({
   ]);
 
   React.useEffect(() => {
+    const sourceEdges = element.getSourceEdges();
     action(() => {
-      const sourceEdges = element.getSourceEdges();
+      const indent = detailsLevel === ScaleDetailsLevel.high && !verticalLayout ? width - pillWidth : 0;
       sourceEdges.forEach(edge => {
         const data = edge.getData();
-        edge.setData({
-          ...(data || {}),
-          indent: detailsLevel === ScaleDetailsLevel.high && !verticalLayout ? width - pillWidth : 0
-        });
+        if ((data?.indent ?? 0) !== indent) {
+          edge.setData({ ...(data || {}), indent });
+        }
       });
     })();
+
+    return action(() => {
+      sourceEdges.forEach(edge => {
+        const data = edge.getData();
+        if (data.indent) {
```

Comment:
```suggestion
        if (data?.indent) {
```

Needs optional chaining here to prevent app crash on resize